main.tf

```diff
@@ -8,7 +8,7 @@ resource "aws_kms_key" "backup" {
 }
 
 resource "aws_kms_alias" "backup" {
-  name          = "alias/aws_backup-${var.vault_name}"
+  name          = var.kms_key_alias == null ? var.kms_key_alias : "alias/aws_backup-${var.vault_name}-${var.env}"
```

Comment:
you have default set for var. kms_key_alias asa empty string and here you check for null, is this ok?